lib/mvos/model/user_model.dart

```diff
@@ -1,34 +1,65 @@
 import 'dart:async';
-
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/foundation.dart';
 import 'package:playground_app/logger/logger.dart';
 import 'package:playground_app/mvos/model/entity/user_entity.dart';
 import 'package:playground_app/mvos/model/observable/user_observable.dart';
 import 'package:playground_app/mvos/service/firebase_service.dart';
 import 'package:playground_app/shared/interfaces.dart';
+import 'package:rxdart/subjects.dart';
+
+typedef Future<void> LoginA({String email, String password});
+typedef Future<void> CreateAccountA(
+    {String name,
+    String surname,
+    String email,
+    String phone,
+    List courseCode});
 
 class UserModel implements Disposable {
   final FirebaseService firebaseService;
 
-  StreamController<LoggedInO> _loggedInO$ = StreamController.broadcast();
-  Stream<LoggedInO> get loggedInO$ => _loggedInO$.stream;
+  BehaviorSubject<LoggedInO> loggedInO$ = BehaviorSubject<LoggedInO>();
+  LoginO loginO;
+  AccountO accountO;
+
   UserModel({
     @required this.firebaseService,
   }) : assert(firebaseService != null,
             'you must initialize the model with firebaseService') {
     _initUserModel();
   }
 
-  @override
-  Future<void> dispose() {
-    _loggedInO$.close();
+  Future<void> login({String email, String password}) async {
+    firebaseService.loginWithEmailAndPassword(email, password);
```

Comment:
```
return     firebaseService.loginWithEmailAndPassword(email, password);
```

Reply:
to je future, i ako baci error oces da ui dobije error, da ne navigira na novu stranicu

Reply:
ok